### PR TITLE
[ALLUXIO-2455] Keystone v3 authentication for Swift UFS

### PR DESCRIPTION
**Pull request by madanadit**
Add support for authentication using Keystone V3 credentials.

JIRA: https://alluxio.atlassian.net/browse/ALLUXIO-2455

Reference: OpenStack Keystone V3 Authentication https://developer.openstack.org/api-ref/identity/v3/?expanded=password-authentication-with-scoped-authorization-detail

### CODE REVIEW

**Comment by madanadit** [comment]
@calvinjia @ifcharming Please take a look. Thanks!

**Comment by madanadit** [comment]
Jenkins, test this please

File: underfs/swift/src/main/java/alluxio/underfs/swift/SwiftUnderFileSystem.java [hunk 1]
```diff
@@ -127,6 +127,14 @@ public SwiftUnderFileSystem(AlluxioURI uri, UnderFileSystemConfiguration conf)
               config.setPreferredRegion(conf.getValue(PropertyKey.SWIFT_REGION_KEY));
             }
             break;
+          case Constants.SWIFT_AUTH_KEYSTONE_V3:
+            if (conf.containsKey(PropertyKey.SWIFT_REGION_KEY)) {
+              config.setPreferredRegion(conf.getValue(PropertyKey.SWIFT_REGION_KEY));
+            }
+            config.setAuthenticationMethod(AuthenticationMethod.EXTERNAL);
+            KeystoneV3AccessProvider accessProvider = new KeystoneV3AccessProvider(config);
```
Comment (calvinjia) [inline underfs/swift/src/main/java/alluxio/underfs/swift/SwiftUnderFileSystem.java]: The library does not provide an access provider for keystone v3?

File: docs/en/Configuring-Alluxio-with-Swift.md [hunk 2]
```diff
@@ -43,7 +43,7 @@ The following configuration should be provided in the `conf/alluxio-site.propert
 {% include Configuring-Alluxio-with-Swift/several-configurations.md %}
 
 Possible values of `<swift-use-public>` are `true`, `false`. Possible values of `<swift-auth-model>`
-are `keystone`, `tempauth`, `swiftauth`. When using `keystone` authentication, the following
+are `keystonev3`, `keystone`, `tempauth`, `swiftauth`. When using `keystone` authentication, the following
```
Comment (calvinjia) [inline docs/en/Configuring-Alluxio-with-Swift.md]: using either keystone authentication?

File: underfs/swift/src/main/java/alluxio/underfs/swift/SwiftUnderFileSystem.java (hunk 1 above)
Comment (madanadit) [inline underfs/swift/src/main/java/alluxio/underfs/swift/SwiftUnderFileSystem.java]: @calvinjia nope, JOSS does not: https://github.com/javaswift/joss/issues/112

File: underfs/swift/src/main/java/alluxio/underfs/swift/KeystoneV3AccessProvider.java [hunk 3]
```diff
@@ -0,0 +1,267 @@
...
+      } catch (JsonProcessingException e) {
+        LOG.error("Error processing JSON request: {}", e.getMessage());
+        return null;
+      }
+
+      HttpURLConnection connection = null;
+      BufferedReader bufReader = null;
+      try {
+        // Send request
+        connection = (HttpURLConnection) new URL(mAccountConfig.getAuthUrl()).openConnection();
```
Comment (jsimsa) [inline underfs/swift/src/main/java/alluxio/underfs/swift/KeystoneV3AccessProvider.java]: correct, I think we should be consistent in our code base and use Apache HttpClient (see https://www.mkyong.com/java/how-to-send-http-request-getpost-in-java/ for reference)

**Pull request merged**